homeassistant/components/nightscout/sensor.py

```diff
@@ -31,21 +32,24 @@ async def async_setup_entry(
 ) -> None:
     """Set up the Glucose Sensor."""
     api = hass.data[DOMAIN][entry.entry_id]
-    async_add_entities([NightscoutSensor(api, "Blood Sugar", entry.unique_id)], True)
+    async_add_entities([NightscoutSensor(api, entry.unique_id)], True)
 
 
 class NightscoutSensor(SensorEntity):
     """Implementation of a Nightscout sensor."""
 
+    _attr_has_entity_name = True
+    _attr_translation_key = "blood_sugar"
     _attr_native_unit_of_measurement = "mg/dL"
     _attr_icon = "mdi:cloud-question"
+    _attr_available = False
 
-    def __init__(self, api: NightscoutAPI, name, unique_id) -> None:
+    def __init__(self, api: NightscoutAPI, unique_id) -> None:
         """Initialize the Nightscout sensor."""
         self.api = api
         self._attr_unique_id = unique_id
-        self._attr_name = name
         self._attr_extra_state_attributes: dict[str, Any] = {}
+        self._attr_device_info = DeviceInfo(identifiers={(DOMAIN, unique_id)})
```

Comment:
I also just saw that I am creating a new device here, while the code is already doing that in `__init__.py` @marciogranzotto, can you give some insight on what would be a nice solution?